Assets Editor/DatEditor.xaml.cs

```diff
@@ -436,122 +436,212 @@ private void LoadCurrentObjectAppearances()
-            A_FlagDescription.Text = CurrentObjectAppearance.HasDescription ? CurrentObjectAppearance.Description : null;
-            A_FlagWrap.IsChecked = CurrentObjectAppearance.Flags.HasWrap;
-            A_FlagUnwrap.IsChecked = CurrentObjectAppearance.Flags.HasUnwrap;
-            A_FlagDecoItemKit.IsChecked = CurrentObjectAppearance.Flags.HasDecoItemKit;
-            A_FlagTopeffect.IsChecked = CurrentObjectAppearance.Flags.HasTop;
-            A_FlagChangedToExpire.IsChecked = CurrentObjectAppearance.Flags.Changedtoexpire != null;
-            A_FlagChangedToExpireId.Value = (CurrentObjectAppearance.Flags.Changedtoexpire != null && CurrentObjectAppearance.Flags.Changedtoexpire.HasFormerObjectTypeid) ? (int)CurrentObjectAppearance.Flags.Changedtoexpire.FormerObjectTypeid : 0;
-            A_FlagCorpse.IsChecked = CurrentObjectAppearance.Flags.Corpse;
-            A_FlagCyclopedia.IsChecked = CurrentObjectAppearance.Flags.Cyclopediaitem != null;
-            A_FlagCyclopediaItem.Value = (CurrentObjectAppearance.Flags.Cyclopediaitem != null && CurrentObjectAppearance.Flags.Cyclopediaitem.HasCyclopediaType) ? (int)CurrentObjectAppearance.Flags.Cyclopediaitem.CyclopediaType : 0;
-            A_FlagAmmo.IsChecked = CurrentObjectAppearance.Flags.HasAmmo;
-            A_FlagShowOffSocket.IsChecked = CurrentObjectAppearance.Flags.ShowOffSocket;
-            A_FlagReportable.IsChecked = CurrentObjectAppearance.Flags.Reportable;
-            A_FlagReverseAddonEast.IsChecked = CurrentObjectAppearance.Flags.ReverseAddonsEast;
-            A_FlagReverseAddonWest.IsChecked = CurrentObjectAppearance.Flags.ReverseAddonsWest;
-            A_FlagReverseAddonNorth.IsChecked = CurrentObjectAppearance.Flags.ReverseAddonsNorth;
-            A_FlagReverseAddonSouth.IsChecked = CurrentObjectAppearance.Flags.ReverseAddonsSouth;
-            A_FlagWearOut.IsChecked = CurrentObjectAppearance.Flags.Wearout;
-            A_FlagClockExpire.IsChecked = CurrentObjectAppearance.Flags.Clockexpire;
-            A_FlagExpire.IsChecked = CurrentObjectAppearance.Flags.Expire;
-            A_FlagExpireStop.IsChecked = CurrentObjectAppearance.Flags.Expirestop;
-            A_FlagUpgradeClassification.IsChecked = CurrentObjectAppearance.Flags.Upgradeclassification != null;
-            A_FlagUpgradeClassificationAmount.Value = (CurrentObjectAppearance.Flags.Upgradeclassification != null && CurrentObjectAppearance.Flags.Upgradeclassification.HasUpgradeClassification) ? (int)CurrentObjectAppearance.Flags.Upgradeclassification.UpgradeClassification : 0;
-
-            NpcDataList.Clear();
-
-            if (CurrentObjectAppearance.Flags.Npcsaledata.Count > 0)
-            {
-                A_FlagNPC.IsChecked = true;
-                foreach (var npcdata in CurrentObjectAppearance.Flags.Npcsaledata)
-                    NpcDataList.Add(npcdata);
+            if (CurrentObjectAppearance == null || CurrentObjectAppearance.Flags == null)
```

Comment:
Thanks for your contribution!
I see you've added a default value for non-existing flags. In the original code, we initialize the object to create these flags when they are absent.  is there a specific scenario you had in mind for this change?

Reply:
Sometimes, when we convert an old spr/dat to assets, these variables might come as null, which was causing crashes (nullptr). So, in this PR, I thought of checking if this variable is null and assigning a default value. Some assets I tested were also missing these flags, which caused errors when trying to open them. But if you think it's unnecessary, I can remove it.

Reply:
And i think it was a strategy to protect the sprites against people dont know how use assets editor in dev mode..

Reply:
Yes, it's not necessary, would be great if you removed this part

Reply:
Removed